esp-hal/src/gpio/mod.rs

```diff
@@ -565,6 +565,23 @@ impl From<usize> for GpioRegisterAccess {
 }
 
 impl GpioRegisterAccess {
+    pub const BANK_COUNT: usize = 1 + cfg!(any(esp32, esp32s2, esp32s3)) as usize;
```

Comment:
doesn't matter but we could derive `EnumCount` and use  `COUNT` - then we wouldn't need to repeat the condition but generally we might want to define the banks via esp-metadata? We have that `cfg` a lot here and when adding new targets it's somewhat annoying to change all of them